pkg/datapath/tables/node_address.go

```diff
@@ -436,12 +476,13 @@ func (n *nodeAddressController) getAddressesFromDevice(dev *Device) sets.Set[Nod
 		index := len(addrs)
 
 		isPublic := ip.IsPublicAddr(addr.Addr.AsSlice())
-		primary := false
 		if addr.Addr.Is4() {
-			if !ipv4Found {
-				ipv4Found = true
-				primary = true
+			if addr.Addr.Unmap() == n.k8sIPv4.Unmap() {
```

Comment:
The `Unmap` seems unnecessary if both IPs are confirmed to be IPv4 addresses

Reply:
What would you propose doing instead? If I remove these the tests fail as one of these ends up being an IPv4-mapped IPv6 address (either MustParseAddr or AddrFromSlice causes it, probably the latter...).

Reply:
> If I remove Unmap the tests fail as one of these ends up being an IPv4-mapped IPv6 address

That is odd, the branch is guarded by an `if addr.Addr.Is4() {` check so `addr.Addr` should be an IPv4 for sure. Which means that we allow IPv4-mapped-IPv6 addresses as valid value for `n.k8sIPv4`.

If we truly support that then there is nothing wrong with the code. I just didn't expected that